html/Makefile

```diff
@@ -1,15 +1,53 @@
 SHELL  := bash
+dyn_json := recipes.json archive.json download_counts.json build-status.json
+pages  := $(notdir $(wildcard templates/pages/*.html))
+p_json := $(wildcard packages/*.json)
+p_html := $(p_json:.json=.html)
+htmldir := $(notdir $(CURDIR))
 
-.PHONY: all clean updates.rss
-all: updates.rss
+all: updates.rss pages packages
 
-updates.rss: updates.rss.erb
-	erb updates.rss.erb > updates.rss.tmp && mv updates.rss.tmp updates.rss
+updates.rss: templates/updates.rss.erb packages.json
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi templates/updates.rss.erb > updates.rss.tmp && mv updates.rss.tmp updates.rss
 
-livejson:
-	for f in recipes archive download_counts build-status; do curl https://melpa.org/$$f.json > $$f.json; done
+%.html: templates/pages/%.html templates/layout.html.erb
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi content_template=$< templates/layout.html.erb > $@.tmp && mv $@.tmp $@
+
+index.html: templates/pages/index.html templates/layout.html.erb total_downloads.json packages.json
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi content_template=$< templates/layout.html.erb > $@.tmp && mv $@.tmp $@
+
+packages/%.html: packages/%.json templates/package.html.erb templates/layout.html.erb
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi pname=$* content_template=templates/package.html.erb templates/layout.html.erb > $@.tmp && mv $@.tmp $@
+
+packages/%.json: packages.json
```

Comment:
Do we need this just for search?

Reply:
No, every page needs its own JSON file since the parsing takes quite a lot of time given the number of packages